velox/functions/prestosql/tests/RepeatTest.cpp

```diff
@@ -0,0 +1,119 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class RepeatTest : public FunctionBaseTest {
+ protected:
+  void testRepeat(
+      const VectorPtr& expected,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<ArrayVector>("repeat(C0, C1)", makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  void testError(
+      const std::vector<VectorPtr>& input,
+      const std::string& expectedError) {
+    VELOX_ASSERT_THROW(
+        evaluate<ArrayVector>("repeat(C0, C1)", makeRowVector(input)),
+        expectedError);
+  }
+};
+
+} // namespace
+
+TEST_F(RepeatTest, repeatInteger) {
+  auto elementVector =
+      makeNullableFlatVector<int32_t>({10, -20, 30, 40, std::nullopt, 50});
+  auto countVector =
+      makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4, std::nullopt});
+  auto expected = makeNullableArrayVector<int32_t>({
+      {{10}},
+      {{-20, -20}},
+      {{30, 30, 30}},
+      {{}},
+      {{std::nullopt, std::nullopt, std::nullopt, std::nullopt}},
+      std::nullopt,
+  });
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, repeatFloat) {
+  auto elementVector = makeNullableFlatVector<float>(
+      {0.0, -2.0, 3.333333, 4.0004, std::nullopt, 5.12345});
+  auto countVector =
+      makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4, std::nullopt});
+  auto expected = makeNullableArrayVector<float>({
+      {{0.0}},
+      {{-2.0, -2.0}},
+      {{3.333333, 3.333333, 3.333333}},
+      {{}},
+      {{std::nullopt, std::nullopt, std::nullopt, std::nullopt}},
+      std::nullopt,
+  });
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, repeatBoolean) {
+  auto elementVector =
+      makeNullableFlatVector<bool>({true, false, true, false, std::nullopt});
+  auto countVector = makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4});
+  auto expected = makeNullableArrayVector<bool>(
+      {{true},
+       {false, false},
+       {true, true, true},
+       {},
+       {std::nullopt, std::nullopt, std::nullopt, std::nullopt}});
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, repeatString) {
+  using S = StringView;
+
+  auto elementVector = makeNullableFlatVector<StringView>(
```

Comment:
Please add a test with large strings > 12 chars to test the out of line storage.